### PR TITLE
feat: Add four new endpoints - Connections, Environments, Repositories, Users

### PR DESCRIPTION
**Pull request by mjsqu**
Data originating from the existing endpoints can be enhanced by adding four new endpoints.

For example, the Jobs endpoint returns records such as:

```json
{
    "type": "RECORD",
    "stream": "jobs",
    "record": {
        "generate_docs": false,
        "id": 590,
        "account_id": 1,
        "project_id": 1,
        "environment_id": 2,
        "name": "ODS HPI Prd",
        "dbt_version": null,
        "execute_steps": [
            "dbt build -s +dwh_ods.hpi",
            "dbt run-operation upload_dbt_artifacts --args '{filenames: [run_results]}'"
        ],
```

Where `environment_id` corresponds to an environment in dbt Cloud. It would be useful if the Environments endpoint were able to be queried, as this would return the corresponding Environment with `id = 2`, giving environment name and other useful values:

```json
{
    "type": "RECORD",
    "stream": "environments",
    "record": {
        "id": 2,
        "account_id": 1,
        "repository_id": 8,
        "created_by_id": null,
        "name": "uat",
        "use_custom_branch": true,
        "custom_branch": "develop",
        "dbt_version": "1.3.0-latest",
        "supports_docs": true,
        "state": 1
    },
    "time_extracted": "2023-05-25T01:35:50.070266+00:00"
}
```

further, the environment record contains `repository_id`, so it could be useful to have the `RepositoriesStream`, bringing through the record:

```json
{
    "type": "RECORD",
    "stream": "repositories",
    "record": {
        "id": 8,
        "account_id": 1,
        "remote_url": "xxxx",
        "remote_backend": "gitlab",
        "git_clone_strategy": "xxx",
        "deploy_key_id": xxxx,
        "github_installation_id": xxxxx,
        "created_at": "2021-08-12T22:50:40.694931+00:00",
        "updated_at": "2021-08-12T22:50:40.694946+00:00",
        "state": 1
    },
    "time_extracted": "2023-05-25T01:41:47.658076+00:00"
}
```

To conclude, I went through the open_api schema available at:
https://raw.githubusercontent.com/fishtown-analytics/dbt-cloud-openapi-spec/master/openapi-v3.yaml

And added any endpoint I could find that either:

- has a `Response` object
- has `get` tag in the `paths:` tag
- is at the `accountId` level and no further Ids appear on the endpoint URI (e.g. not things like: `/accounts/{accountId}/runs/{runId}/artifacts/`)

and crucially, the new endpoints that I have added I was able to successfully query using the modified tap with at least one API key :)

404 warning - Some endpoints are only available to privileged users, and the API appears to handle this at our site by return 404 if the endpoint is not reachable. I don't know how to handle this error properly in the Meltano SDK - the fix for me would be to "not select" the stream, e.g. I have had varying success querying `{accountId}/repositories/` with different API keys.

### CODE REVIEW

**Comment by mjsqu** [comment]
Hang on - fixing and adding tests...

**Comment by mjsqu** [comment]
pre-commit.ci autofix

**Comment by mjsqu** [comment]
This is done as far as I can manage - unsure how to resolve the 8 function arguments down to < 5 to pass the final pre-commit check, may need a bit of a refactor.

**Comment by tayloramurphy** [comment]
Thanks for the PR! I've tagged @edgarrmondragon to take a look when he has some time.

File: tests/test_core.py [hunk 1]
```diff
@@ -178,12 +301,141 @@ def runs_response():
...
+                "avatar_url": None,
+                "fullname": faker.name(),
+                "show_existing_user_email_verification": False,
+            },
+        ],
+    }
+
+
 @responses.activate
 def test_standard_tap_tests(
```
Comment (edgarrmondragon) [inline tests/test_core.py:430]:
```suggestion
def test_standard_tap_tests(  # noqa: PLR0913
```

**Review by edgarrmondragon — APPROVED** [review]
We can add a `noqa` just for that error. We should indeed refactor, but I don't think this is a blocker.

File: tests/test_core.py [hunk 2]
```diff
@@ -19,6 +19,10 @@
     "account_ids": ["1000"],
 }
 
+def fake_date(faker: Faker):
```
Comment (edgarrmondragon) [inline tests/test_core.py:22]:
```suggestion

def fake_date(faker: Faker):
```

**Comment by edgarrmondragon** [comment]
Thanks @mjsqu! I'll cut a release later today 🙂

**Pull request merged**